diff --git a/src/components/Modal/Modal.jsx b/src/components/Modal/Modal.jsx
index 91843d0..f9b069b 100644
--- a/src/components/Modal/Modal.jsx
+++ b/src/components/Modal/Modal.jsx
@@ -57,7 +57,7 @@ export class SimpleModal extends React.Component {
               </Typography>
               <Typography variant="subtitle2">
                 <b>Genres: </b>
-                {`${genres.join(', ')}`}
+                {/* {`${genres.join(', ')}`} */}
               </Typography>
             </div>
           </div>
diff --git a/src/components/MovieList/MovieListItem/MovieListItem.jsx b/src/components/MovieList/MovieListItem/MovieListItem.jsx
index a9536c7..cb0b729 100644
--- a/src/components/MovieList/MovieListItem/MovieListItem.jsx
+++ b/src/components/MovieList/MovieListItem/MovieListItem.jsx
@@ -57,7 +57,7 @@ export class MovieListItem extends React.Component {
                   {overview}
                 </Typography>
                 <Typography variant="body2" color="textSecondary">
-                  {`Genres: ${genres.join(', ')}`}
+                  {/* {`Genres: ${genres.join(', ')}`} */}
                 </Typography>
               </Grid>
             </Grid>
diff --git a/src/constants/actions.js b/src/constants/actions.js
index efd7c92..710c9da 100644
--- a/src/constants/actions.js
+++ b/src/constants/actions.js
@@ -1,3 +1,5 @@
 export const MOVIES = {
-  SEARCH: 'GET_SEARCH_MOVIES',
+  SEARCH: 'SEARCH_MOVIES',
+  SEARCH_RESET: 'SEARCH_MOVIES_RESET',
+  LOAD: 'LOAD_MOVIES',
 };
diff --git a/src/containers/MoviePage.js b/src/containers/MoviePage.js
index 900d6df..4ba2f72 100644
--- a/src/containers/MoviePage.js
+++ b/src/containers/MoviePage.js
@@ -1,8 +1,9 @@
 import { connect } from 'react-redux';
 import MoviePage from '../page/MoviePage';
 
-const mapStateToProps = state => ({
-  ...state.payload,
+const mapStateToProps = ({ search, moviePage }) => ({
+  ...moviePage,
+  movies: search.movies.length ? search.movies : moviePage.movies,
 });
 
 const mapDispatchToProps = dispatch => ({
diff --git a/src/containers/Search.js b/src/containers/Search.js
index 8f3f75d..e1d4c36 100644
--- a/src/containers/Search.js
+++ b/src/containers/Search.js
@@ -2,7 +2,7 @@ import { connect } from 'react-redux';
 import { Search } from '../components/Search/Search';
 
 const mapStateToProps = state => ({
-  ...state,
+  ...state.search,
 });
 
 const mapDispatchToProps = dispatch => ({
diff --git a/src/page/MoviePage.jsx b/src/page/MoviePage.jsx
index 6be4511..82637c4 100644
--- a/src/page/MoviePage.jsx
+++ b/src/page/MoviePage.jsx
@@ -2,6 +2,7 @@ import React from 'react';
 import PropTypes from 'prop-types';
 
 import { PAGE_COUNT } from '../constants/constants';
+import { MOVIES } from '../constants/actions';
 import Search from '../containers/Search';
 import MovieList from '../components/MovieList/MovieList';
 
@@ -9,18 +10,8 @@ import { getMovies } from '../api/api';
 import { FlatPagination } from '../components/FlatPagination/FlatPagination';
 
 export default class MoviePage extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      movies: [],
-      page: 1,
-      totalResults: 0,
-      showCircular: true,
-    };
-  }
-
   componentDidMount() {
-    const { page } = this.state;
+    const { page } = this.props;
     this.loadMovies(page);
   }
 
@@ -38,21 +29,29 @@ export default class MoviePage extends React.Component {
   }
 
   loadMovies(page) {
-    const { type } = this.props;
+    const { type, getActionDispatcher } = this.props;
     return getMovies(type, page)
       .then(({ movies, totalResults }) => {
-        this.setState({ movies, totalResults, page });
-      })
-      .then(() => {
-        this.setState({ showCircular: false });
+        getActionDispatcher({
+          type: MOVIES.LOAD,
+          payload: {
+            movies,
+            totalResults,
+            page,
+            showCircular: false,
+          },
+        })();
+        getActionDispatcher({
+          type: MOVIES.SEARCH_RESET,
+        })();
       });
   }
 
   render() {
-    const { title } = this.props;
     const {
-      movies, page, totalResults, showCircular,
-    } = this.state;
+      title, movies, page, totalResults, showCircular,
+    } = this.props;
+
     return (
       <div>
         <FlatPagination
diff --git a/src/reducers/moviePage.js b/src/reducers/moviePage.js
index b96d94e..76ac528 100644
--- a/src/reducers/moviePage.js
+++ b/src/reducers/moviePage.js
@@ -1,12 +1,22 @@
 import { MOVIES } from '../constants/actions';
 
-export const search = (state = {}, action) => {
+const defaultState = {
+  movies: [],
+  page: 1,
+  totalResults: 0,
+  showCircular: true,
+};
+
+export const moviePage = (state = defaultState, action) => {
   switch (action.type) {
-    case MOVIES.SEARCH:
-      return action.payload;
+    case MOVIES.LOAD:
+      return {
+        ...defaultState,
+        ...action.payload,
+      };
     default:
       return state;
   }
 };
 
-export default search;
+export default moviePage;
diff --git a/src/reducers/search.js b/src/reducers/search.js
index b96d94e..5d80802 100644
--- a/src/reducers/search.js
+++ b/src/reducers/search.js
@@ -1,9 +1,17 @@
 import { MOVIES } from '../constants/actions';
 
-export const search = (state = {}, action) => {
+const defaultState = {
+  movies: [],
+};
+
+export const search = (state = defaultState, action) => {
   switch (action.type) {
     case MOVIES.SEARCH:
-      return action.payload;
+      return {
+        ...action.payload,
+      };
+    case MOVIES.SEARCH_RESET:
+      return defaultState;
     default:
       return state;
   }
